docs/DeveloperGuide.md

```diff
@@ -275,6 +275,83 @@ The following activity diagram summarizes what happens when a user executes an a
+* `MainWindow` - The UI component that handles navigating through fields.
+* `CommandType` - The Enum class that represents the type of command which MainWindow checks to handle the UI change.
+
+Given below is an example usage scenario and how the mechanism behaves at each step, given that the user already opened person profile page:
+
+Step 1. The user enters the name of the field. e.g. "name". Since the normal person list is invisible, "name" is passed to `executeInView()` method in `MainWindow` class.
+
+![EditFieldSequenceDiagramStep1.png](images/EditFieldSequenceDiagramStep1.png)
+
+Step 2. With `executeInView()`, `ViewModeParser` is used to parse the command which returns `EditFieldCommand`.
+
+![EditFieldSequenceDiagramStep2.png](images/EditFieldSequenceDiagramStep2.png)
+
+Step 3. `EditFieldCommand` is executed, and with the `CommandType.EDIT_FIELD` carried by `CommandResult`, `MainWindow` calls `handleEditField()` method. The interaction between `MainWindow` and `PersonProfile` is covered in detail in **[View Feature](#view-feature)**. 
+
+![EditFieldSequenceDiagram.png](images/EditFieldSequenceDiagram.png)
+
+<br>
+
+### Saving the Changes in Profile Page Feature 
+
+The mechanism allows the user to save the edited details of a fosterer in their profile page. This feature is facilitated by `ViewModeParser`, and `SaveCommand` classes, to handle user input in the profile page and save the updated fosterer. This feature is implemented using the following components and operations:
+
+* `ViewModeParser` - Represents the parser that parses commands that are executed in a fosterer's profile.
+* `SaveCommand` - The core component responsible for saving the changes made by the user.
+* `MainWindow` - The UI component that handles navigating through fields.
+* `CommandType` - The Enum class that represents the type of command which MainWindow checks to handle the UI change.
+
+Given below is an example usage scenario and how the save mechanism behaves at each step, given that the user already opened person profile page:
+
+Step 1. The user enters `save` command. Since the normal person list is invisible, the command text "save" is passed to `executeInView()` method in `MainWindow` class. 
+
+![SaveSequenceDiagramStep1.png](images/SaveSequenceDiagramStep1.png)
+
+Step 2. With `executeInView()`, `ViewModeParser` is used to parse the command text which returns `SaveCommand`.
+
+![SaveSequenceDiagramStep2.png](images/SaveSequenceDiagramStep2.png)
+
+Step 3. `EditFieldCommand` is executed, and `setPerson()` method from `Model` class is called with personToEdit and targetIndex obtained from `MainWindow` class.
+
+![SaveSequenceDiagram.png](images/SaveSequenceDiagram.png) 
+
+Step 4. From `MainWindow`, `handleSave()` handler method is called which calls `resetValues()` in `PersonProfile` class that updates the field values to the currently saved fosterer's details and change the text color from red, if exists, back to black.
+
```

Comment:
Please update instructions for manual testing to include some tests for ur features!!